comit/src/network/orderbook.rs

```diff
@@ -80,6 +80,8 @@ impl Orderbook {
             events: VecDeque::new(),
         };
 
+        std::thread::sleep(std::time::Duration::from_secs(2));
```

Comment:
@rishflab just pointed me to this debug-PR.

when trying to connect taker-ui (behind a cnd node) and nectar I ran into the problem that Nectar never properly registers the taker once we dial.

In a setup with 2 cnd nodes (one for the taker and one for the maker) when the taker dials the maker, we see the registration of the taker on the maker's side:

```
Jul 30 14:41:20.065  INFO libp2p_gossipsub::behaviour: New peer connected: PeerId("12D3KooWJdBwGAohCJbaZDFurkFkVkLuXCQmSa7wJ7cVMgujLJRR")    
Jul 30 14:41:20.130  INFO libp2p_gossipsub::behaviour: SUBSCRIPTION: Adding peer: PeerId("12D3KooWJdBwGAohCJbaZDFurkFkVkLuXCQmSa7wJ7cVMgujLJRR") to topic: TopicHash { hash: "BTC/DAI" } 
```

However, when using Nectar we never see this - the subscription is somehow not done.
The incoming connection is noticed in Nectar (when actively monitoring `libp2p::swarm::SwarmEvent::IncomingConnection` events), but we subscription is never done,

Thought I report this here.

Reply:
Thanks for the report @da-kami.